crates/shared/src/db_order_conversions.rs

```diff
@@ -54,6 +54,14 @@ pub fn full_order_into_model_order(order: database::orders::FullOrder) -> Result
         sender: onchain_user,
         placement_error: onchain_placement_error,
     });
+    let full_fee_amount =
+        big_decimal_to_u256(&order.full_fee_amount).context("full_fee_amount is not U256")?;
+    let solver_fee = match &class {
+        OrderClass::Limit(limit) => limit.surplus_fee.context("missing surplus_fee")?,
+        OrderClass::Market => full_fee_amount,
+        OrderClass::Liquidity => full_fee_amount,
+    };
```

Comment:
Not really sure what to put here (`full_fee_amount` or `fee_amount`).
Some of the earlier code suggested to use `full_fee_amount` and would have used the subsidized amount (i.e.`surplus_fee` or `fee_amount`). I think this was a bug so it's probably good that this value gets initialized only once now.
Also different values were used for liquidity vs regular orders which didn't make sense to me.

Reply:
Intuitively, I would have expected this to be:
```rust
    let solver_fee = match &class {
        OrderClass::Limit(limit) => full_fee_amount,
        OrderClass::Market => full_fee_amount,
        OrderClass::Liquidity => big_decimal_to_u256(&order.fee_amount).context("...")?,
    };
```

The reason here is:
- Full fee amount represents the fee that we actually think the order will use to settle. This is what we want to use for objective value computation (otherwise a subsidy will make the order count less towards an solution's objective value despite having the same complexity - notably, CoW token holder subsidized orders are less attractive). This is the same for limit orders as `surplus_fee` contains subsidies that should not be deducted from the `solver_fee`.
- Liquidity orders should use `fee_amount` (and I think they should probably not get scaled at all). The rationale here is that their fee doesn't contribute to the objective value unless they bribe the protocol with fees.